src/api/comm/packers/IntegerPacker.cpp

```diff
@@ -0,0 +1,14 @@
+#include <array>
+#include "DataPacker.h"
+
+namespace twister::comm::packers {
+    class IntegerPacker : public DataPacker<int> {
+
+        void pack_to_buffer(int data, Buffer buffer, PackerStatus status) {
```

Comment:
Think Buffer and PackerStatus should be passed as reference here?
`void pack_to_buffer(int data, Buffer **&** buffer, PackerStatus **&** status) {`

If we don't need to make accidental copies of Buffer object in any case we can ask compiler to delete the copy ctor, and Copy assignment ctors also. 
`
Buffer(const Buffer &bf) = delete;
Buffer& operator=(const Buffer&) = delete;
`